packages/connection/src/class-error-handler.php

```diff
@@ -0,0 +1,568 @@
+		$parsed_token = explode( ':', wp_unslash( $token ) );
+
+		if ( isset( $parsed_token[2] ) && ! empty( $parsed_token[2] ) && ctype_digit( $parsed_token[2] ) ) {
+			$user_id = $parsed_token[2];
+		} else {
+			$user_id = 'invalid';
+		}
+
+		return $user_id;
+
+	}
+
+	/**
+	 * Gets the reported errors stored in the database
+	 *
+	 * @since 8.7.0
+	 *
+	 * @return array $errors
+	 */
+	public function get_stored_errors() {
+
+		$stored_errors = get_option( self::STORED_ERRORS_OPTION );
+
+		if ( ! is_array( $stored_errors ) ) {
+			$stored_errors = array();
+		}
+
+		$stored_errors = $this->garbage_collector( $stored_errors );
+
+		return $stored_errors;
+	}
+
+	/**
+	 * Gets the verified errors stored in the database
+	 *
+	 * @since 8.7.0
+	 *
+	 * @return array $errors
+	 */
+	public function get_verified_errors() {
+
+		$verified_errors = get_option( self::STORED_VERIFIED_ERRORS_OPTION );
+
+		if ( ! is_array( $verified_errors ) ) {
+			$verified_errors = array();
+		}
+
+		$verified_errors = $this->garbage_collector( $verified_errors );
+
+		return $verified_errors;
+	}
+
+	/**
+	 * Removes expired errors from the array
+	 *
+	 * This method is calleb by get_stored_errors and get_verified errors and filters their result
+	 * Whenever a new error is stored to the database or verified, this will be triggered and the
+	 * expired error will be permantently removed from the database
+	 *
+	 * @since 8.7.0
+	 *
+	 * @param array $errors array of errors as stored in the database.
+	 * @return array
+	 */
+	private function garbage_collector( $errors ) {
+		foreach ( $errors as $error_code => $users ) {
+			foreach ( $users as $user_id => $error ) {
+				if ( self::ERROR_LIFE_TIME < time() - (int) $error['timestamp'] ) {
+					unset( $errors[ $error_code ][ $user_id ] );
+				}
+			}
+		}
+		// Clear empty error codes.
+		$errors = array_filter(
+			$errors,
+			function( $user_errors ) {
+				return ! empty( $user_errors );
+			}
+		);
+		return $errors;
+	}
+
+	/**
+	 * Delete all stored and verified errors from the database
+	 *
+	 * @since 8.7.0
+	 *
+	 * @return void
+	 */
+	public function delete_all_errors() {
+		$this->delete_stored_errors();
+		$this->delete_verified_errors();
+	}
+
+	/**
+	 * Delete the reported errors stored in the database
+	 *
+	 * @since 8.7.0
+	 *
+	 * @return boolean True, if option is successfully deleted. False on failure.
+	 */
+	public function delete_stored_errors() {
+		return delete_option( self::STORED_ERRORS_OPTION );
+	}
+
+	/**
+	 * Delete the verified errors stored in the database
+	 *
+	 * @since 8.7.0
+	 *
+	 * @return boolean True, if option is successfully deleted. False on failure.
+	 */
+	public function delete_verified_errors() {
+		return delete_option( self::STORED_VERIFIED_ERRORS_OPTION );
+	}
+
+	/**
+	 * Gets an error based on the nonce
+	 *
+	 * Receives a nonce and finds the related error. If error is found, move it to the verified errors option.
```

Comment:
The error isn't moved to the verified options in this method.

```suggestion
	 * Receives a nonce and finds the related error.
```